src/kOS/Control/ThrottleManager.cs

```diff
@@ -0,0 +1,118 @@
+﻿using kOS.Safe.Encapsulation;
+using kOS.Safe.Exceptions;
+using kOS.Safe.Utilities;
+using System;
+
+namespace kOS.Control
+{
+    public class ThrottleManager : IFlightControlParameter
+    {
+        private Vessel internalVessel;
+        private uint controlPartId;
+        private SharedObjects controlShared;
+
+        public bool Enabled { get; private set; }
+        public double Value { get; set; }
+
+        public ThrottleManager(Vessel vessel)
+        {
+            Enabled = false;
+            controlPartId = 0;
+
+            internalVessel = vessel;
+        }
+
+        uint IFlightControlParameter.ControlPartId
+        {
+            get
+            {
+                return controlPartId;
+            }
+        }
+
+        bool IFlightControlParameter.Enabled
+        {
+            get
+            {
+                return Enabled;
+            }
+        }
+
+        bool IFlightControlParameter.IsAutopilot
+        {
+            get
+            {
+                return true;
+            }
+        }
+
+        void IFlightControlParameter.CopyFrom(IFlightControlParameter origin)
+        {
+            object val = origin.GetValue();
+            Value = Convert.ToDouble(val);
+        }
+
+        void IFlightControlParameter.DisableControl()
+        {
+            controlShared = null;
+            controlPartId = 0;
+            Enabled = false;
+        }
+
+        void IFlightControlParameter.DisableControl(SharedObjects shared)
+        {
+            if (shared.KSPPart.flightID == controlPartId)
```

Comment:
@Dunbaratu found out why you were seeing the control act funny.  The code is returning if the currently controlling part **is** the part calling to disable, instead of returning if the part calling for disable **is not** currently controlling the manager.  The same issue is in wheel steer and wheel throttle.  I will modify accordingly.

Reply:
I was trying to read through the code to find the problem (because, hey, it's a good way to force me to understand the code for code review reasons), and I noticed this but didn't feel confident enough in my understanding of what you were doing here to raise a flag until I dug deeper.  It *felt* like it didn't make sense and would prevent the unlock from working most of the time, but I didn't know what you were *trying* to do here so I didn't want to say outright that it was wrong.  It didn't occur to me to think, "Would it have made sense if it was inverted?  Maybe that was what he meant.".